### DISCUSSION TITLE
How to remove reference branches from the left panel?

### DISCUSSION THREAD

**Discussion by codenjs** [gitextensions/gitextensions#8478] [General]
I have a remote with dozens of branches. At some point, I fetched many of them but I only use a few. They take up too much space in the left panel, so I want to remove some. If I right click on them, I see a Delete option, but I don't want to delete anything in the remote. I only want to remove my local reference. Is that possible? It isn't clear what the Delete option will do.

**Comment by vbjay** [comment]
Look in the plugins menu.

**Reply by codenjs** [reply]
Which plugin are you referring to? I see a plugin option "Delete obsolete branches" but that doesn't look like it solves my problem

**Comment by RussKie** [comment]
Apologies, but I think you may be confused a little.

**Branches** node contains your local branches, which may or may not have related remote branches. Deleting these will delete your local branches.
**Remotes** node contains your remotes _and_ all remote branches a given remote has. Deleting these will delete remotes.

You can remove merged local branches either manually or with the help of the "Delete obsolete branches" plugin.
You can "fetch and prune" that will remove locally cached remote branches that no longer exist on a remote.

**Reply by codenjs** [reply]
I think my confusion was about the difference between branches and refs. With some more searching, I found the command that solves my problem: git update-ref -d

It removes my local reference to a remote branch without modifying anything in the remote repo. Now my UI only shows me the branches I care about, regardless of how many branches actually exist in the remote repo.

**Comment by vbjay** [comment]
Yes. The obsolete branches plugin is what I was pointing you to. Based on
what you tell it, you could use it to do what you wanted.
